### PR TITLE
Script for automating cleanup issue creation for deprecated featur…

### PR DESCRIPTION
**Pull request by htuch**
…es at a given version

Usage:

tools/deprecate_version.py <deprecate for version> <deprecate by version>

e.g

tools/deprecate_version.py 1.6.0 1.7.0

A GitHub access token must be set in GH_ACCESS_TOKEN. To create one, go to
Settings -> Developer settings -> Personal access tokens in GitHub and create
a token with public_repo scope. Keep this safe, it's broader than it needs to
be thanks to GH permission model
(https://github.com/dear-github/dear-github/issues/113).

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

**Comment by htuch** [comment]
@mattklein123 @alyssawilk as promised. Here's the issues it created: https://github.com/envoyproxy/envoy/issues?utf8=%E2%9C%93&q=label%3Adeprecation+

The limitations of this approach can be seen in the spurious issue created at https://github.com/envoyproxy/envoy/issues/2903. I think maybe we could do something a bit smarter to avoid missing PRs (a recursive `git blame` of sorts), but it's never going to be as precise as manual tagging.

If you think this is worth continuing with, I can cleanup the PR. I actually think this is a useful automation example for where we want to go with some of our future bots, so we could treat it just as a reference / learning exercise, I'm cool either way.

**Comment by alyssawilk** [comment]
Very cool!!  Will take a look tomorrow if it isn't merged by then.

I think whoever creates these killing spurious ones is way better than our current system of having to track things manually.

minor nit "This issue will track source code cleanup." -> "This issue tracks source code cleanup"

And we totally want it to auto-tag!  And auto-assign!  And and and.... launch and iterate, right.
thanks so much for an awesome first pass :-D

**Comment by alyssawilk** [comment]
@envoyproxy/maintainers

File: tools/deprecate_version.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+set -e
+
+SCRIPT_DIR=$(realpath "$(dirname "$0")")
+
+mkdir -p build
+cd build
+
+# TODO(htuch): avoid creating virtualenv every time
+virtualenv deprecate_version
```
Comment (mattklein123) [inline tools/deprecate_version.sh]: nit: can we do a proper requirements.txt here for the venv?

Comment (htuch) [reply tools/deprecate_version.sh]: Sure. @danielhochman let me clean this up before diving into the Python minutiae, I'll ping you when it's ready.

**Comment by alyssawilk** [comment]
I'd lean towards running it when we cut a release, so the issues get filed when folks should work on them.  Alternately if we want to have tracking issues filed early we could tag them or some such and make pinging them part of the release process. I just think that many of the folks who get these won't be following enough to know when we cut a release and want to optimize for the issues not getting lost :-)

File: tools/deprecate_version.sh (hunk 1 above)
Comment (htuch) [reply tools/deprecate_version.sh]: @danielhochman ready for review.

**Comment by danielhochman** [comment]
lgtm @htuch

**Pull request merged**